sdks/java/container/license_scripts/pull_licenses_java.py

```diff
@@ -140,17 +149,22 @@ def execute(dep):
     }
     '''
 
-    name = dep['moduleName'].split(':')[1].lower()
+    name = dep['moduleName'].split(':')[1]
     version = dep['moduleVersion']
     name_version = name + '-' + version
+    # javac is not a runtime dependency
+    if name == 'javac':
+      logging.debug('Skipping', name_version)
+      return
     dir_name = '{license_dir}/{name_version}.jar'.format(
         license_dir=license_dir, name_version=name_version)
 
     # if auto pulled, directory is existing at {license_dir}
     if not os.path.isdir(dir_name):
         # skip self dependencies
         if dep['moduleName'].startswith('beam'):
-            logging.debug('Skippig', name_version)
+            logging.debug('Skipping', name_version)
```

Comment:
Do we only skip self-dependencies for non-auto-pulled libraries? Mostly this function seems very long and complicated to start with, and having the skipping returns at the top would make me feel a little better.

Reply:
Moved to the top.